components/Messages/Messages.js

```diff
@@ -0,0 +1,96 @@
+import * as Styles from './Messages.styles'
+import Typography from '@material-ui/core/Typography'
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+import { Button } from '@material-ui/core'
+
+export default function Messages(props) {
+	const {
+		messages,
+		onOpenMessage: openMessage,
+		onDeleteMessage: deleteMessage,
+	} = props
+
+	console.log(messages)
+
+	const [viewMode, setViewMode] = useState({
+		edit: false,
+	})
+
+	const toggleEditMode = () => {
+		setViewMode({
+			edit: !viewMode.edit,
+		})
+		resetSelectedState()
+	}
+
+	const [selectedMessages, setSelectedMessages] = useState([])
```

Comment:
This is a lot of logic in this component. This looks like it should be broken into redux state where you can have all of this tested.